src/main/kotlin/org/arend/projectView/ArendProjectViewStructureProvider.kt

```diff
@@ -0,0 +1,115 @@
+package org.arend.projectView
+
+import com.intellij.icons.AllIcons
+import com.intellij.ide.projectView.*
+import com.intellij.ide.projectView.impl.nodes.NamedLibraryElementNode
+import com.intellij.ide.projectView.impl.nodes.PsiFileNode
+import com.intellij.ide.util.treeView.AbstractTreeNode
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.LibraryOrderEntry
+import com.intellij.openapi.roots.impl.libraries.LibraryEx
+import com.intellij.openapi.vfs.VirtualFile
+import org.arend.ext.module.ModulePath
+import org.arend.module.AREND_LIB
+import org.arend.module.ArendLibraryKind
+import org.arend.module.ArendRawLibrary
+import org.arend.psi.ArendFile
+import org.arend.typechecking.TypeCheckingService
+import org.arend.util.FileUtils
+
+class ArendProjectViewStructureProvider : TreeStructureProvider {
+    override fun modify(parent: AbstractTreeNode<*>,
+                        children: MutableCollection<AbstractTreeNode<*>>,
+                        settings: ViewSettings?)
+            : MutableCollection<AbstractTreeNode<*>> {
+        if (parent !is NamedLibraryElementNode) {
+            return children
+        }
+        val library = findArendLibrary(parent)
+        if (library == null || library.config.additionalModulesSet.isEmpty()) {
+            return children
+        }
+        return mutableListOf<AbstractTreeNode<*>>().apply {
+            addAll(children)
+            add(ArendMetasNode(parent.project, library, settings))
+        }
```

Comment:
> I think in general you don't want to commit the .arc files into the repo

Indeed, discussed that with Valery, need to find a way to avoid those bin files.

> If you don't care about the order you could try listOf(ArendMetasNode(parent.project, library, settings), *children)

True, sorting is performed later, I can use `listOf`. Will do, thanks for the suggestions)